src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -457,27 +457,21 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 // We only support string and enum keys
                 return;
             }
-
+            MethodInfo tryGetValue = dictionaryType.GetMethod("TryGetValue")!;
             PropertyInfo setter = dictionaryType.GetProperty("Item", DeclaredOnlyLookup)!;
             foreach (IConfigurationSection child in config.GetChildren())
             {
+                var key = keyTypeIsEnum ? Enum.Parse(keyType, child.Key) : child.Key;
```

Comment:
```suggestion
                object key = keyTypeIsEnum ? Enum.Parse(keyType, child.Key) : child.Key;
```
We don't use `var` unless it is obvious from the right hand side.